BigQuery/tests/System/ManageTablesTest.php

```diff
@@ -347,4 +347,93 @@ private function runJob($jobConfig, $client = null)
 
         $this->assertArrayNotHasKey('errorResult', $job->info()['status']);
     }
+
+    /**
+     * @dataProvider referenceFileSchemaTestUris
+     */
+    public function testCreateTableWithReferenceFileSchemaUri($sourceUris, $refUri, $format)
+    {
+        $id = uniqid(self::TESTING_PREFIX);
+        $options = [
+            'externalDataConfiguration' => [
+                'sourceUris' => $sourceUris,
+                'referenceFileSchemaUri' => $refUri,
+                'sourceFormat' => $format
+            ]
+        ];
+
+        $table = self::$dataset->createTable($id, $options);
+
+        $this->assertTrue(self::$dataset->table($id)->exists());
+        $this->assertEquals($id, $table->id());
+
+        $columns = $table->info()['schema']['fields'];
+        $i = 0;
+        foreach ($columns as $column) {
+            $this->assertEquals(self::REFERENCE_SCHEMA[$i++], $column['name']);
+        }
+    }
+
+    /**
+     * @dataProvider referenceFileSchemaTestUris
+     */
+    public function testLoadTableWithReferenceFileSchemaUri($sourceUris, $refUri, $format)
+    {
+        $id = uniqid(self::TESTING_PREFIX);
+        $table = self::$dataset->table($id);
+        $loadConfig = $table->load('');
+        $loadConfig->sourceUris($sourceUris);
+        $loadConfig->referenceFileSchemaUri($refUri);
+        $loadConfig->sourceFormat($format);
+        $loadConfig->destinationTable($table);
+
+        $job = self::$client->runJob($loadConfig);
+
+        $this->assertTrue(self::$dataset->table($id)->exists());
+        $this->assertEquals($id, $table->id());
+
+        $columns = $table->info()['schema']['fields'];
+        $i = 0;
+        foreach ($columns as $column) {
+            $this->assertEquals(self::REFERENCE_SCHEMA[$i++], $column['name']);
+        }
+    }
+
+    /**
+     * @dataProvider referenceFileSchemaTestUris
+     */
+    public function testQueryTableWithReferenceFileSchemaUri($sourceUris, $refUri, $format)
+    {
+        $id = uniqid(self::TESTING_PREFIX);
+        $queryConfig = self::$client->query(sprintf('SELECT * FROM %s', $id));
+        $queryConfig->tableDefinitions([$id =>[
+            'sourceUris' => $sourceUris,
+            'referenceFileSchemaUri' => $refUri,
+            'sourceFormat' => $format
+        ]]);
+        $result = self::$client->runQuery($queryConfig);
+        $rows = $result->rows();
+
+        foreach ($rows as $row) {
+            $keys = array_keys($row);
+            break;
+        }
+        $this->assertEquals(self::REFERENCE_SCHEMA, $keys);
```

Comment:
Okay, Actually the table will have only one schema and that is already checked outside the loop. Even if we move that check inside, I don't think it will account to much time, so, we can move the schema check to inside. If thats moved inside, we need that greater than zero check as it is possible that no rows are being retrieved.